sdk/core/core-http/test/policies/throttlingRetryPolicyTests.ts

```diff
@@ -114,6 +114,28 @@ describe("ThrottlingRetryPolicy", () => {
       assert.deepEqual(response, mockResponse);
     });
 
+    it("should pass the response to the handler if the status code equals 503", async () => {
+      const request = new WebResource();
+      const mockResponse = {
+        status: 503,
+        headers: new HttpHeaders({
+          "Retry-After": "100"
+        }),
+        request: request
+      };
+      const policy = createDefaultThrottlingRetryPolicy(mockResponse, (_, response) => {
+        delete (response.request as any).requestId;
+        delete (mockResponse.request as any).requestId;
+        assert.deepEqual(response, mockResponse);
+        return Promise.resolve(response);
+      });
+
+      const response = await policy.sendRequest(request);
+      delete (request as any).requestId;
+      delete (response.request as any).requestId;
+      assert.deepEqual(response, mockResponse);
+    });
+
```

Comment:
since we've had so many bugs in this space, can we have tests that it 1) retries repeated 503/429 and 2) does not exceed the max retry count when doing so?